src/plugins/intel_cpu/src/cpu_streams_calculation.cpp

```diff
@@ -30,48 +30,68 @@ std::vector<std::vector<int>> get_streams_info_table(const int input_streams,
+                    stream_info[THREADS_PER_STREAM] = proc_type_table[0][n];
+                    streams_info_table.push_back(stream_info);
+                    n_threads -= proc_type_table[0][n];
+                }
+            }
+        }
+    };
+
     if (1 == input_streams) {
         stream_info[NUMBER_OF_STREAMS] = 1;
-        int limit_threads = (input_threads == 0) ? model_prefer_threads : input_threads;
-        if (proc_type_table[0][ALL_PROC] == proc_type_table[0][EFFICIENT_CORE_PROC]) {
-            stream_info[PROC_TYPE] = EFFICIENT_CORE_PROC;
-            stream_info[THREADS_PER_STREAM] = (input_threads == 0)
-                                                  ? proc_type_table[0][EFFICIENT_CORE_PROC]
-                                                  : std::min(proc_type_table[0][EFFICIENT_CORE_PROC], limit_threads);
-            streams_info_table.push_back(stream_info);
-        } else if ((proc_type_table[0][EFFICIENT_CORE_PROC] > 0) &&
-                   ((limit_threads == 0) || (limit_threads > proc_type_table[0][MAIN_CORE_PROC]))) {
-            stream_info[PROC_TYPE] = ALL_PROC;
-            int n_threads = std::accumulate(proc_type_table[0].begin() + MAIN_CORE_PROC,
-                                            proc_type_table[0].begin() + HYPER_THREADING_PROC,
-                                            0);
-            stream_info[THREADS_PER_STREAM] = (limit_threads == 0) ? n_threads : std::min(n_threads, limit_threads);
-            streams_info_table.push_back(stream_info);
-            stream_info[NUMBER_OF_STREAMS] = 0;
-            n_threads = stream_info[THREADS_PER_STREAM];
-            for (int n = MAIN_CORE_PROC; n < HYPER_THREADING_PROC; n++) {
-                if (0 != proc_type_table[0][n]) {
-                    stream_info[PROC_TYPE] = n;
-                    if (n_threads <= proc_type_table[0][n]) {
-                        stream_info[THREADS_PER_STREAM] = n_threads;
-                        streams_info_table.push_back(stream_info);
-                        break;
-                    } else {
-                        stream_info[THREADS_PER_STREAM] = proc_type_table[0][n];
-                        streams_info_table.push_back(stream_info);
-                        n_threads -= proc_type_table[0][n];
-                    }
-                }
+        if (input_threads > 0) {
+            stream_info[THREADS_PER_STREAM] = std::min(proc_type_table[0][ALL_PROC], input_threads);
+            if ((stream_info[THREADS_PER_STREAM] > proc_type_table[0][MAIN_CORE_PROC]) &&
+                (proc_type_table[0][MAIN_CORE_PROC] > 0) && (proc_type_table[0][EFFICIENT_CORE_PROC] > 0)) {
+                stream_info[PROC_TYPE] = ALL_PROC;
+                streams_info_table.push_back(stream_info);
+                UpdateMixStreamInfo();
+            } else if ((stream_info[THREADS_PER_STREAM] <= proc_type_table[0][MAIN_CORE_PROC]) ||
+                       (proc_type_table[0][EFFICIENT_CORE_PROC] == 0)) {
```

Comment:
Is there such a situation as `stream_info[THREADS_PER_STREAM] > proc_type_table[0][MAIN_CORE_PROC] && proc_type_table[0][EFFICIENT_CORE_PROC] == 0` which should uses logical core?

Reply:
Yes. For one stream using both physical core and logic core of Pcore, the core type of stream will be physical core of Pcore, and number of threads in stream will be total number of physical core and logic core.